src/plots/cartesian/select.js

```diff
@@ -553,7 +553,7 @@ function isOnlyThisBinSelected(searchTraces, clickedPtInfo) {
         if(isSameTrace) {
             trace = clickedPtInfo.searchInfo.cd[0].trace;
             if(trace.selectedpoints.length === clickedPtInfo.pointNumbers.length) {
-                for(i = 0; i > clickedPtInfo.pointNumbers.length; i++) {
+                for(i = 0; i < clickedPtInfo.pointNumbers.length; i++) {
```

Comment:
Looking at the code again, it would have been caught by a Jasmine test. However, the calling code in the if condition in `selectOnClick` was "covering" the bug due to an additional call to `isPointOrBinSelected`. This is somewhat redundant and I'll simply the if condition appropriately.